### DISCUSSION TITLE
How to get project root dir

### DISCUSSION THREAD

**Discussion by gagara** [neoclide/coc.nvim#3502] [Q&A] [answered]
Is there any way to get root dir of current project (the one which is send to language server)?
Thanks.

**Comment by fannheyward** [comment, marked as answer]
Try `:CocCommand workspace.workspaceFolders`.

**Reply by gagara** [reply]
I tried. But it returns the list of all opened projects. But I need to know the project for which currently opened file belongs to.
Maybe it is possible to pass some extra arguments to it?

**Reply by changyiyao** [reply]
I just has the same request. 
waiting for the reply.

**Reply by gagara** [reply]
Having the list of all projects dirs and full file path for current buffer it is possible to find "current" project with following function:
```vimscript
function! SetVimtestProjectRoot(file)
	if exists("g:WorkspaceFolders")
		for f in g:WorkspaceFolders
			if match(a:file, f, 0) == 0 | let g:test#project_root = f | return | endif
		endfor
	endif
endfunction
```
Here I assign current project dir to `g:test#project_root` (it is used by `vim-test` plugin). But it is easy to add any other logic.
Also I have autocmd to do it once file is opened:
```vimscript
autocmd BufEnter * :call SetVimtestProjectRoot(expand('%:p'))
```  
Maybe there is better/simpler solution, though?

**Reply by changyiyao** [reply]
Good ideas.
 In my scenario, I use `bottomUp` configuration with `rootPattern` to recognize workspaces. When the above iterable codes should be reverse which is right for me, since buffer to `/a/b/file` should match the most closest path `/a/b` as its workspace, which is the last item in `g:WorkspaceFolders` when I open vim in shell pwd `/a`.
